pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -45,8 +45,21 @@ type BuildRunSpec struct {
 	// image would be pushed to. It will overwrite the output image in build spec
 	// +optional
 	Output *Image `json:"output,omitempty"`
+
+	// State is used for canceling a buildrun (and maybe more later on).
+	// +optional
+	State BuildRunRequestedState `json:"stateOverride,omitempty"`
 }
 
+// BuildRunRequestedState defines the buildrun state the user can provide to override whatever is the current state.
+type BuildRunRequestedState string
+
+const (
+	// BuildRunStateCancel indicates that the user wants to cancel the BuildRun,
+	// if not already canceled or terminated
+	BuildRunStateCancel = "BuildRunCanceled"
```

Comment:
```suggestion
	BuildRunStateCancel = "Canceled"
```
This seems cleaner and I think it is still obvious what it means.

Reply:
Maybe it would be useful to also define a few other of the `state` options now also?
```
BuildRunStateRunning = "Running"
...
```

Reply:
no in this case I want some commonality with how tekton does it

that string will show up in the cli get operations

I want to leave this as is

Reply:
an as v1beta1.TaskRunReasonRunning already produces "Running" and that by default is proxied from the taskrun to the buildrun when updating the buildrun conditions, I don't see value at this time for adding builrunstaterunning